lua_sandbox/executor.py

```diff
@@ -786,15 +793,43 @@ def _from_python(self, cls, val, recursion=0, max_recursion=10):
 
         raise TypeError("Can't serialise %r. Do you need a capsule?" % (val,))
 
+    @contextlib.contextmanager
+    def as_buffer(self):
+        """
+        Yields a buffer() directly into the memory that contains this Lua string.
+
+        This is useful for zero-allocation access to lua strings, but comes at
+        the cost of needing to be very careful with how you affect the Lua VM
+        while you hold onto this reference. The referred string must remain on
+        the Lua stack the whole time you retain a reference to it, you must
+        release that reference before as_buffer() returns, and you must never
+        write to the buffer
+        """
+        with self._bring_to_top():
+            kind = lua_type(self.L, -1)
+            if kind != _executor.LUA_TSTRING:
+                raise TypeError("as_buffer only works on strings!")
 
-def _callable_wrapper(executor, val):
+            size = ctypes.c_size_t(0)
+            ptr = lua_tolstring(self.L, -1, ctypes.byref(size))
+            # size-1 because Lua includes room for a null byte but we don't
+            # need it
+            cls = (ctypes.c_char * size.value)
+            buff = cls.from_address(ptr)
+            yield buff
```

Comment:
Can you invalidate `buff` after this yield? That would make escaped references (but not derived pointers into the underlying buffer!) somewhat safer.

Reply:
Alternatively, this could check buff's refcount and get grumpy if it's not exactly 1. (or 2 for sys.getrefcount's argument reference). That would force the callers to `del` or otherwise remove references.